service/src/main/kotlin/no/nav/su/se/bakover/service/søknadsbehandling/SøknadsbehandlingServiceImpl.kt

```diff
@@ -167,18 +165,27 @@ internal class SøknadsbehandlingServiceImpl(
         val søknadsbehandling = søknadsbehandlingRepo.hent(request.behandlingId)
             ?: return KunneIkkeBeregne.FantIkkeBehandling.left()
 
-        return statusovergang(
-            søknadsbehandling = søknadsbehandling,
-            statusovergang = Statusovergang.TilBeregnet {
-                BeregningStrategyFactory(clock).beregn(
-                    grunnlagsdataOgVilkårsvurderinger = GrunnlagsdataOgVilkårsvurderinger(
-                        grunnlagsdata = søknadsbehandling.grunnlagsdata,
-                        vilkårsvurderinger = søknadsbehandling.vilkårsvurderinger,
-                    ),
-                    beregningsPeriode = søknadsbehandling.periode, begrunnelse = request.begrunnelse,
-                )
-            },
-        ).let {
+        return søknadsbehandling.beregn(
+            begrunnelse = request.begrunnelse,
+            clock = clock,
+        ).getOrHandle {
+            return when (it) {
+                is Søknadsbehandling.KunneIkkeBeregne.UgyldigTilstand -> {
+                    KunneIkkeBeregne.UgyldigTilstand(it.fra, it.til)
+                }
+                is Søknadsbehandling.KunneIkkeBeregne.UgyldigTilstandForEndringAvFradrag -> {
+                    KunneIkkeBeregne.UgyldigTilstandForEndringAvFradrag(it.feil.toService())
+                }
+                Søknadsbehandling.KunneIkkeBeregne.AvkortingErUfullstendig -> {
+                    KunneIkkeBeregne.AvkortingErUfullstendig
+                }
+            }.left()
+        }.let {
+            // må lagre fradrag på nytt, siden eventuelle avkortinger legges til ved beregning
+            grunnlagService.lagreFradragsgrunnlag(
+                behandlingId = it.id,
+                fradragsgrunnlag = it.grunnlagsdata.fradragsgrunnlag,
```

Comment:
Ligger avkortinger også i fradragsgrunnlagene? Hvordan ser det ut i frontend? Blir det masse ekstrahåndtering pgr. det?

Reply:
I det øyeblikk man ender opp med  å avkorte beløp vil disse legges til beregninge på samme måte som andre fradrag. Håndteringen i frontend er ganske lik den vi har for f.eks forventet inntekt, hvor saksbehandler ikke får muligheten til å legge til fradrag av denne typen. Ellers ikke noe spesiell håndtering.